### ISSUE TITLE
Getting invalid count value errors intermittently during deployment

### ISSUE THREAD

**Issue by asnaseer-resilient** [aws/aws-cdk#8667]
I intermittently see errors reported below this during my stack deployments.
The error I have shown is an example from trying to deploy a lambda function but I also see this during deployments of RDS and AppSync. This never occurred on the previous version of the CDK that I was running (**v1.39.0**)

### Reproduction Steps
I use CircleCI to deploy my stacks. A typical deployment command looks like this:
```
cdk deploy dbSchemaMigrator --app .build/stacks/cfd-api-stacks.js -o .build/cdk.out/dbSchemaMigrator --no-staging --exclusively --require-approval never
```

Not sure if this is relevant as I never saw this error in the older release of the CDK.

When my CircleCI build fails with the above error, I can see that the actual AWS stack has been created successfully so I suspect there is an error in the stack progress monitoring code in the latest release. Re-running my CircleCI job always succeeds.

### Error Log

```
dbSchemaMigrator (cfd-api-dbSchemaMigrator-uat): deploying...
[0%] start: Publishing a747fb3608b7a2e29cc9cd7a740f4df1d5081a0cea693982c96991f1a3ed5e25:current
[100%] success: Published a747fb3608b7a2e29cc9cd7a740f4df1d5081a0cea693982c96991f1a3ed5e25:current
cfd-api-dbSchemaMigrator-uat: creating CloudFormation changeset...
 ❌  dbSchemaMigrator (cfd-api-dbSchemaMigrator-uat) failed: RangeError: Invalid count value
    at String.repeat (<anonymous>)
    at CurrentActivityPrinter.progressBar (/home/circleci/project/node_modules/aws-cdk/lib/api/util/cloudformation/stack-activity-monitor.ts:558:24)
    at CurrentActivityPrinter.print (/home/circleci/project/node_modules/aws-cdk/lib/api/util/cloudformation/stack-activity-monitor.ts:494:23)
    at StackActivityMonitor.flushEvents (/home/circleci/project/node_modules/aws-cdk/lib/api/util/cloudformation/stack-activity-monitor.ts:151:18)
    at StackActivityMonitor.stop (/home/circleci/project/node_modules/aws-cdk/lib/api/util/cloudformation/stack-activity-monitor.ts:106:12)
    at processTicksAndRejections (internal/process/task_queues.js:94:5)
    at Object.deployStack (/home/circleci/project/node_modules/aws-cdk/lib/api/deploy-stack.ts:271:7)
    at CdkToolkit.deploy (/home/circleci/project/node_modules/aws-cdk/lib/cdk-toolkit.ts:181:24)
    at main (/home/circleci/project/node_modules/aws-cdk/bin/cdk.ts:252:16)
    at initCommandLine (/home/circleci/project/node_modules/aws-cdk/bin/cdk.ts:184:9)
Invalid count value
```

### Environment

  - **CLI Version      : v1.46.0**
  - **Node.js Version: circleci/node:12.14.1**
  - **OS               : CircleCI node_executor**
  - **Language (Version): Typescript 3.9.5**

### Other
I also see this error intermittently when trying to run the deploy script on my local machine (Mac OS Catalina 10.15.5) with Node 12.4.0.
--- 

This is :bug: Bug Report

**Comment by NetaNir** [comment]
You are correct, the errors is thrown from the progress bar visualization code (that was added in version 1.46.0), as you said this does not effect the actual deployment as CloudFormation deployments are asynchronies.

We are on it, sorry for that!

----------------------------------

@rix0rrr the error is probably due to calling `String.repeat` with a negative value here:

 ```typescript  
  const filler = '·'.repeat(width - 2 - Math.floor(chars) - (partialChar ? 1 : 0));
```
I tried some combinations but couldn't find an immediate suspect.
